pkg/splunk/enterprise/configuration.go

```diff
@@ -778,76 +783,125 @@ func AreRemoteVolumeKeysChanged(client splcommon.ControllerClient, cr splcommon.
 	}
 
-	if appFramework.S3Bucket == "" {
-		return fmt.Errorf("Apps Remote Storage S3Bucket is missing")
+	if localScope && appFramework.Defaults.Scope != "" && appFramework.Defaults.Scope != "local" {
+		return fmt.Errorf("Invalid scope for defaults config. Only local scope is supported for this kind of CR")
 	}
 
-	if appFramework.S3SecretRef == "" {
-		return fmt.Errorf("Apps Remote Storage S3SecretRef is missing")
+	if appFramework.Defaults.Scope != "" && appFramework.Defaults.Scope != "local" && appFramework.Defaults.Scope != "cluster" {
+		return fmt.Errorf("Scope for defaults should be either local Or cluster, but configured as: %s", appFramework.Defaults.Scope)
 	}
 
-	if appFramework.S3PollInterval == 0 {
-		// this means either the S3PollInterval was not provided
-		// or it was explictely set to 0 - in both cases we will
-		// set the S3PollInterval to default 60 minutes
-		scopedLog.Info("Setting S3PollInterval to default 60 minutes")
-		appFramework.S3PollInterval = 60
+	if appFramework.Defaults.VolName != "" {
+		_, err := checkIfVolumeExists(appFramework.VolList, appFramework.Defaults.VolName)
+		if err != nil {
+			return fmt.Errorf("Invalid Volume Name for Defaults. Error: %s", err)
+		}
 	}
 
 	return nil
 }
 
-// ValidateSplunkSmartstoreSpec checks and validates the smartstore config
-func ValidateSplunkSmartstoreSpec(smartstore *enterprisev1.SmartStoreSpec) error {
-	var err error
+//  isAppFrameworkConfigured checks and returns true if App Framework is configured
+//  App Repo config without any App sources will not cause any App Framework activity
+func isAppFrameworkConfigured(appFramework *enterprisev1.AppFrameworkSpec) bool {
+	return !(appFramework == nil || appFramework.AppSources == nil)
+}
 
-	// Smartstore is an optional config (at least) for now
-	if !isSmartstoreConfigured(smartstore) {
+// ValidateAppFrameworkSpec checks and validates the Apps Frame Work config
+func ValidateAppFrameworkSpec(appFramework *enterprisev1.AppFrameworkSpec, localScope bool) error {
+	var err error
+	if !isAppFrameworkConfigured(appFramework) {
 		return nil
 	}
 
-	numVolumes := len(smartstore.VolList)
-	numIndexes := len(smartstore.IndexList)
-	if numIndexes > 0 && numVolumes == 0 {
-		return fmt.Errorf("Volume configuration is missing. Num. of indexes = %d. Num. of Volumes = %d", numIndexes, numVolumes)
+	scopedLog := log.WithName("ValidateAppFrameworkSpec").WithValues(
+		"localScope: %t", localScope)
+
+	// Todo: sgontla: Enable later. For now, allowing multiple volumes and app sources for UT purpose
+	if len(appFramework.VolList) > 1 || len(appFramework.AppSources) > 2 {
+		//return fmt.Errorf("Invalid App Framework config. Only one Volume, and a max. of two App sources supported")
+	}
+
+	scopedLog.Info("Validating app framework spec")
+
+	//ToDo: sgontla: Do we need to return error, if the poll interval is too small? TBD
+	if appFramework.AppsRepoPollInterval < minAppsRepoPollInterval {
```

Comment:
Is AppsRepoPollInterval mandatory? If not, by not providing the spec param, can we default (to say 1 hour)?